### PR TITLE
fix hard-coded values in multiplication timeindex and resolution

### LINKED ISSUE #13: Add comprehensive word to magic number converting time index to time
SEE https://github.com/moonshot-nagayama-pj/multiharp-toolkit/pull/9#pullrequestreview-2188375402

1. Add comprehensive word to magic number "5"
2. Make comprehensive function calculating `timestamp * 5`

### PR DESCRIPTION
**Pull request by Sakumatcho**
closes #13 
I named magic number "5", "globRes(resolution)".

Actually, we can get from "MeasDesc_GlobalResolution" when we use ptu_parser.
But, I can not check the actual operation now, so I put a constant value.

### CODE REVIEW

**Comment by Sakumatcho** [comment]
I made issue.
https://github.com/moonshot-nagayama-pj/multiharp-toolkit/issues/12

File: src/multiharp_toolkit/ptu_parser.py [hunk 1]
```diff
@@ -98,12 +99,17 @@ def parse_record(self, data: int):
             truetime = self.oflcorrection + timetag
             self.append_events(channel + 1, truetime)
 
-    def append_events(self, channel: int, timestamp: float):
+    def timetag2time(self, timetag: int) -> int:
```
Comment (auspicacious) [inline src/multiharp_toolkit/ptu_parser.py:102]: Function names should start with verbs, such as "convert." They should not use non-standard grammar like "2" for "to."

"Time" is too vague. What is this value? Is it a relative timestamp?

```suggestion
    def convert_timetag_to_relative_timestamp(self, timetag: int) -> int:
```

File: src/multiharp_toolkit/ptu_parser.py [hunk 2]
```diff
@@ -98,12 +99,17 @@ def parse_record(self, data: int):
...
+        """convert time tag to time(unit: psec)
```
Comment (auspicacious) [inline src/multiharp_toolkit/ptu_parser.py:103]: Avoid abbreviations.

Avoid vagueness.

```suggestion
        """Convert a time tag to a relative timestamp with picosecond resolution.
```

File: src/multiharp_toolkit/ptu_parser.py (hunk 1 above)
Comment (Sakumatcho) [reply src/multiharp_toolkit/ptu_parser.py:102]: Thank you for comment.

>"Time" is too vague. What is this value? Is it a relative timestamp?

Sorry, I am not sure..
I will check.

I think it is relative timestamp, but, I do not know from what the relative timestamp is.

File: src/multiharp_toolkit/ptu_parser.py (hunk 2 above)
Comment (Sakumatcho) [reply src/multiharp_toolkit/ptu_parser.py:103]: I think it is probably good.
However, I want to tell the unit of the number returned from this method has picosecond.
I suppose "timestamp with picosecond resolution" means "this timestamp has resolution of 1 picosecond", however, I do not expect this mean.

What I want to inform in this sentense is , for example, simply 10 means 10 picosecond, and 1000 means 1000 picosecond or 1 nano second.
How should I represent in this case?

Comment (auspicacious) [reply src/multiharp_toolkit/ptu_parser.py:103]: > What I want to inform in this sentense is , for example, simply 10 means 10 picosecond, and 1000 means 1000 picosecond or 1 nano second.
How should I represent in this case?

The word "resolution" usually means "10 means 10 picoseconds". For example, [it's used that way in the Postgres documentation](https://www.postgresql.org/docs/current/datatype-datetime.html). "With picosecond resolution" also has this meaning.

However, we can make this even simpler:

```suggestion
        """Convert a time tag to a relative timestamp in picoseconds.
```

File: src/multiharp_toolkit/ptu_parser.py (hunk 1 above)
Comment (Sakumatcho) [reply src/multiharp_toolkit/ptu_parser.py:102]: I might overlooked it, but I couldn't find the clear definition of "relative timestamp".

>But this is probably good enough for now.

I think so.
So the method name you suggested is good.

File: src/multiharp_toolkit/ptu_parser.py (hunk 2 above)
Comment (Sakumatcho) [reply src/multiharp_toolkit/ptu_parser.py:103]: >The word "resolution" usually means "10 means 10 picoseconds".

I see.

>I suppose "timestamp with picosecond resolution" means "this timestamp has resolution of 1 picosecond", however, I do not expect this mean.
This is my mis-understanding.

So I think first one is good.
Thank you.

**Pull request merged**